core/src/backends/lnd/rest/node.rs

```diff
@@ -96,4 +97,16 @@ impl NodeMethods for LndRest {
 
         Ok(data.try_into()?)
     }
+
+    async fn decode_invoice(&self, invoice: String) -> Result<DecodeInvoiceResult, Error> {
+        let url = format!("{}/v1/payreq/{}", self.config.url, invoice);
+
+        let mut response = self.client.get(&url).send().await?;
+
+        response = Self::on_response(response).await?;
+
+        let data: DecodeInvoiceResponse = response.json().await?;
+
+        Ok(data.into())
```

Comment:
```suggestion
        Ok(data.try_into()?)
```
This should by a `try_into` (`TryInto<DecodeInvoiceResult>` must first be implemented on `DecodeInvoiceResponse`).